golos.publication/objects.hpp

```diff
@@ -102,6 +102,14 @@ struct voteinfo {
     uint64_t by_message() const {
         return message_id;
     }
+
+    uint128_t by_combine_fields() const {
+        return combine_fields(message_id, voter);
```

Comment:
cdt now supports composite keys, there is no need to form uint128. check https://github.com/GolosChain/cyberway.cdt/pull/13 description to see how to use composite keys